plugins/catalog-backend-module-fmeflow/src/FmeFlowEntityProvider.ts

```diff
@@ -68,44 +96,86 @@ export class FMEFlowEntityProvider implements EntityProvider {
+        metadata: {
+          name,
+          namespace: 'default',
+          title: item.title || item.name,
+          description: item.description?.replace(/<\/?[^>]+>/g, '') ?? '',
+          annotations: {
+            [ANNOTATION_LOCATION]: `url:${this.client['baseUrl']}`,
+            [ANNOTATION_ORIGIN_LOCATION]: `url:${this.client['baseUrl']}`,
+            ...(dbRelations?.database && { 'fmeflow/database': dbRelations.database }),
+            ...(dbRelations?.dataset && { 'fmeflow/dataset': dbRelations.dataset }),
+          },
+        },
+        spec: {
+          type: item.type ?? 'fme-workspace',
+          lastUpdated: item.lastPublishDate,
+          ...(dbRelations?.tables?.length && { tables: dbRelations.tables }),
+        },
+      };
+  
+      const cleanDbName = dbRelations?.database
+        ?.toLowerCase()
+        .replace(/[`]/g, '')
+        .replace(/\s+/g, '-')
+        .replace(/[^a-z0-9\-]/g, '');
 
-        const entity: FMEWorkspaceEntity = {
+    const dbEntity: Entity | undefined = cleanDbName
+      ? {
           apiVersion: 'geoportia.se/v1alpha1',
-          kind: 'FMEWorkspace',
+          kind: 'Resource',
           metadata: {
-            name: cleanName,
-            title: item.title ?? cleanName,
-            description: item.description?.replace(/<\/?[^>]+>/g, '') ?? '',
+            name: cleanDbName,
+            title: dbRelations?.database,
+            namespace: 'default',
             annotations: {
               [ANNOTATION_LOCATION]: `url:${this.client['baseUrl']}`,
               [ANNOTATION_ORIGIN_LOCATION]: `url:${this.client['baseUrl']}`,
             },
           },
           spec: {
-            type: item.type ?? 'fme-workspace',
-            lastUpdated: item.lastPublishDate,
-            lifecycle: 'production',
-            owner: 'data-team',
+            type: 'database',
           },
-        };
-
-        if (!isFMEWorkspaceEntity(entity)) {
-          this.logger.warn('Invalid FMEWorkspaceEntity generated: ' + JSON.stringify(entity));
-          return undefined;
         }
-
-        return entity;
-      })
-      .filter((e): e is FMEWorkspaceEntity => e !== undefined);
+      : undefined;
+        
+      const tableEntities: Entity[] =
+        dbRelations?.tables?.map(t => ({
+          apiVersion: 'geoportia.se/v1alpha1',
+          kind: 'Table',
+          metadata: {
+            name: `${t.schema}.${t.table}`.toLowerCase(),
+            namespace: 'default',
+            title: t.table,
+            annotations: {
+              [ANNOTATION_LOCATION]: `url:${this.client['baseUrl']}`,
+              [ANNOTATION_ORIGIN_LOCATION]: `url:${this.client['baseUrl']}`,
+            },
+          },
+          spec: {
+            schema: t.schema,
+            lifecycle: 'production',
+          },
+        })) ?? [];
   
+      return [
+        workspaceEntity,
+        ...(dbEntity ? [dbEntity] : []),
+        ...tableEntities,
+      ];
+    });
```

Comment:
This package should not be producing these entities, as they may overwrite the entities produced by the postgres plugin. I believe it is up to Backstage to handle the case where the remote part of a relation does not exist (yet).